aip/general/0162.md

```diff
@@ -25,198 +25,170 @@ to the version of an API as a whole.
+revisions as nested collection of the resource. Sometimes, the revisions
+collection can be a top level collection, exceptions include:
+
+- If resource revisions are meant to have longer lifespan than the parent
+resource. In other words, resource revisions exist after the resource deletion.
 
-  // The revision ID of the book.
-  // A new revision is committed whenever the book is changed in any way.
-  // The format is an 8-character hexadecimal string.
-  string revision_id = 5 [
-    (google.api.field_behavior) = IMMUTABLE,
-    (google.api.field_behavior) = OUTPUT_ONLY];
+The resource revision type **must** have a field named `snapshot` of the parent
+resource type.
+
+```proto
+message BookRevision {
+   // The name of the book revision.
+   string name = 1;
+   
+   // The snapshot of the book
+   Book snapshot = 2;
 
   // The timestamp that the revision was created.
-  google.protobuf.Timestamp revision_create_time = 6
+  google.protobuf.Timestamp create_time = 3
+    [(google.api.field_behavior) = OUTPUT_ONLY];
+
+  // Other revision IDs that share the same snapshot.
+  repeated string alternate_ids = 4
     [(google.api.field_behavior) = OUTPUT_ONLY];
 }
 ```
 
-- The resource **must** contain a `revision_id` field, which **should** be a
-  string and contain a short, automatically-generated random string. A good
-  rule of thumb is the last eight characters of a [UUID version 4 (random)](UUID4).
-  - The `revision_id` field **must** document when new revisions are created
-    (see [committing revisions](#committing-revisions) below).
-  - The `revision_id` field **should** document the format of revision IDs.
-- The resource **must** contain a `revision_create_time` field, which
-  **should** be a `google.protobuf.Timestamp` (see [AIP-142][]).
-
-**Note:** A randomly generated string is preferred over other methods, such as
-an auto-incrementing integer, because there is often a need to delete or revert
-revisions, and a randomly generated string holds up better in those situations.
+- The resource revision **must** contain a field with a message type of the
+  parent resource, with a field name of the resource singular.
+- The resource revision **must** contain a `create_time` field (see [AIP-142][]).
+- The resource revision **may** contain a repeated field `alternate_ids`, which would
+  contain a list of resource IDs that the revision is also known by (e.g. `latest`)
 
-### Referencing revisions
+### Creating Revisions
 
-When it is necessary to refer to a specific revision of a resource, APIs
-**must** use the following syntax: `{resource_name}@{revision_id}`. For
-example:
+Depending on the resource, different APIs may have different strategies for
+when to create a new revision, such as:
 
-    publishers/123/books/les-miserables@c7cfa2a8
+- Create a new revision any time that there is a change
+- Create a new revision when something important happens
```

Comment:
Nit: "something important happens" isn't concrete enough for me, but that's just me.
```suggestion
- Create a new revision when important system state changes
```